### ISSUE TITLE
`align_double_arrow` fixer sometimes aligns with extra whitespace

### ISSUE THREAD

**Issue by jaredh159** [FriendsOfPHP/PHP-CS-Fixer#1055]
Before:

``` php
$foo = [
    "bar" => 1,
    "baz"   => 2,
];
```

After:

``` php
$foo = [
    "bar"   => 1,
    "baz"   => 2,
];
```

What I would expect:

``` php
$foo = [
    "bar" => 1,
    "baz" => 2,
];
```

Maybe this wasn't the intention of the fixer, so it's not an issue, but I often rename variables in ways that leave double arrows too far to the right, and I was expecting the fixer to use as little whitespace as possible to align them.  I created a failing test case and tried to fix it, but the code in this fixer is a little over my head.

**Comment by keradus** [comment]
@dericofilho what do you think ?

**Comment by GrahamCampbell** [comment]
I mentioned this issue before. If we have way too many spaces, the fixer won't strip them. It'll just align everything really far over.

**Comment by dericofilho** [comment]
`align_double_arrow` is meant to align to the rightmost `T_DOUBLE_ARROW`. Back then, when we debated about that, I argued that this fixer should be somehow tied to another fixer (that, AFAIK, does not exist in PHP-CS-Fixer) that would do space normalization first.

In the case of php.tools, I do have a Pass (a.k.a Fixer, here) that normalize spaces between tokens before actually aligning `T_DOUBLE_ARROW` or `=`.

So it is a schrodinger bug. It is and it is not wrong.

**Comment by nochso** [comment]
Currently working around this by fixing twice:
First with `unalign_double_arrow` and again with `align_double_arrow`. Thus stripping extra spaces before aligning with the minimum offset.

I've tried using both fixers at once, but that didn't work for me. Is there some other way without running php-cs-fixer twice?

**Comment by SpacePossum** [comment]
if we make the prio of `unalign_double_arrow` < of `again with align_double_arrow` then running the fixer with both fixers might do the trick ;)

**Comment by GrahamCampbell** [comment]
> I've tried using both fixers at once, but that didn't work for me. Is there some other way without running php-cs-fixer twice?

You MUST NOT enable those fixers at the same time. They are conflicting. In 2.0, we're working on a way for php-cs-fixer to actually stop people doing that.

**Comment by keradus** [comment]
@GrahamCampbell is right

**Comment by keradus** [comment]
Closes due to merge of #2573
